android/app/src/main/java/com/ecomap/ecomap/data/UserStore.kt

```diff
@@ -16,25 +16,38 @@ class UserStore(private val context: Context) {
         private val Context.dataStore by preferencesDataStore(name = "user")
         private val USER_TOKEN = stringPreferencesKey("user_token")
     }
-    
+
     /**
      * Retrieves user JWT token.
      * @return Coroutine with JWT token or `null` when JWT token is not found.
      */
-    fun getToken(): Flow<String?> {
-        return context.dataStore.data
-            .map { preferences ->
-                preferences[USER_TOKEN]
-            }
+    fun getToken(): String? {
+        return runBlocking {
```

Comment:
We are better of using a `SharedPreferences` if the operations on the data store are always synchronous

Reply:
Aren't `SharedPreferences` being deprecated?